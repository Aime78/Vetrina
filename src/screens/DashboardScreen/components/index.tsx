import Welcome from './Welcome/Welcome';
import Configura from './Configura/Configura';
import Visitors from './Visitors/Visitors';
import Orders from './Orders/Orders';
import Latest from './Latest/Latest';
import Marketplace from './Marketplace/Marketplace';
import Captera from './Captera/Captera';

export {Welcome, Configura, Visitors, Orders, Latest, Marketplace, Captera};
