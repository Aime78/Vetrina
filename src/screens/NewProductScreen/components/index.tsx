import AllProductsButton from './AllProductsButton/AllProductsButton';
import GeneralInfo from './GeneralInfo/GeneralInfo';
import Price from './Price/Price';
import Details from './Details/Details';
import ProductType from './ProductType/ProductType';

export {AllProductsButton, GeneralInfo, Price, Details, ProductType};
