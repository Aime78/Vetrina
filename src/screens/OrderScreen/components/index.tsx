import OrderHeader from './OrderHeader/OrderHeader';
import Cliente from './Cliente/Cliente';
import ShippingInfo from './ShippingInfo/ShippingInfo';

export {OrderHeader, Cliente, ShippingInfo};
