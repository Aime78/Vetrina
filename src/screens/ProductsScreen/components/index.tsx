import HeaderProducts from './HeaderProducts/HeaderProducts';
import TableProducts from './TableProducts/TableProducts';
export {HeaderProducts, TableProducts};
