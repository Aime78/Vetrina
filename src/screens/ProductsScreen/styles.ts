import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: '100%',
    backgroundColor: '#f1f5f9',
    paddingHorizontal: 20,
    paddingTop: 20,
    paddingBottom: 60,
  },
});

export default styles;
