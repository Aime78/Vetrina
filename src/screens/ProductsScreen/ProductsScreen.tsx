import {View} from 'react-native';
import {HeaderProducts, TableProducts} from './components';
import styles from './styles';

const ProductsScreen = () => (
  <View style={styles.container}>
    <HeaderProducts />
    <TableProducts />
  </View>
);

export default ProductsScreen;
