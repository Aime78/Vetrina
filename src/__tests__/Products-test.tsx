import React from 'react';
import {HeaderProducts, TableProducts} from 'screens/ProductsScreen/components';
import renderer from 'react-test-renderer';

it('renders the header of the products screen appropriately', () => {
  const tree = renderer.create(<HeaderProducts />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the table of the products screen appropriately', () => {
  const tree = renderer.create(<TableProducts />).toJSON();
  expect(tree).toMatchSnapshot();
});
