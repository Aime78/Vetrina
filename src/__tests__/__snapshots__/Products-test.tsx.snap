// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the header of the products screen appropriately 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#546679",
          "fontSize": 13,
          "fontWeight": "400",
          "marginBottom": 20,
        }
      }
    >
      Products ( 24 / 100 )
    </Text>
    <Text
      style={
        {
          "color": "#546679",
          "fontSize": 13,
          "fontWeight": "400",
          "marginBottom": 20,
        }
      }
    >
      Featured products (7/10)
    </Text>
    <Image
      source={
        {
          "testUri": "../../../src/assets/switch.png",
        }
      }
    />
  </View>
  <View>
    <Image
      source={
        {
          "testUri": "../../../src/assets/button-plus.png",
        }
      }
    />
  </View>
</View>
`;

exports[`renders the table of the products screen appropriately 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 530,
        "marginTop": 30,
        "width": 370,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      },
    ]
  }
>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#233B53",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Product name
      </Text>
    </View>
    <View
      style={
        {
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.25,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#233B53",
            "fontSize": 17,
            "fontWeight": "600",
          }
        }
      >
        Price
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "#E7E9EC",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 0.6,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/t-shirt.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          T shirt
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderLeftColor": "#E7E9EC",
            "borderLeftWidth": 1,
            "flex": 0.25,
            "padding": 10,
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 14,
            "fontWeight": "400",
          }
        }
      >
        30
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderLeftColor": "#E7E9EC",
          "borderLeftWidth": 1,
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/dots.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 30,
        "height": 60,
        "paddingLeft": 35,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#6C7C8C",
            "fontSize": 10,
            "fontWeight": "600",
          }
        }
      >
        P
      </Text>
      <Text
        style={
          {
            "color": "#6C7C8C",
            "fontSize": 10,
            "fontWeight": "600",
          }
        }
      >
        25
      </Text>
      <Image
        source={
          {
            "testUri": "../../../src/assets/chevron-down.png",
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          {
            "color": "#6C7C8C",
            "fontSize": 10,
            "fontWeight": "600",
          }
        }
      >
        1-25 of 25
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 15,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/chevron-left.png",
          }
        }
      />
      <Image
        source={
          {
            "testUri": "../../../src/assets/chevron-right.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
